identity/registry/provider_registrar.go

```diff
@@ -170,27 +186,41 @@ func (pr *ProviderRegistrar) handleEvent(qe queuedEvent) error {
 }
 
 func (pr *ProviderRegistrar) registerIdentity(qe queuedEvent) error {
+	id := identity.FromAddress(qe.event.ProviderID)
+
 	if !pr.cfg.IsTestnet2 {
-		eligible, err := pr.txer.CheckIfRegistrationBountyEligible(identity.FromAddress(qe.event.ProviderID))
+		eligible, err := pr.txer.CheckIfRegistrationBountyEligible(id)
 		if err != nil {
-			log.Error().Err(err).Msgf("eligibility for registration check failed for %q", qe.event.ProviderID)
+			log.Error().Err(err).Msgf("eligibility for registration check failed for %q", id.Address)
 			return errors.Wrap(err, "could not check eligibility for auto-registration")
 		}
 
 		if !eligible {
-			log.Info().Msgf("provider %q not eligible for auto registration, will require manual registration", qe.event.ProviderID)
+			log.Info().Msgf("provider %q not eligible for auto registration, will require manual registration", id.Address)
+			return nil
+		}
+	}
+
+	// This part migrates bounty payout address to BC beneficiary
+	payout, err := pr.mysteriumAPI.GetPayoutInfo(id, pr.signerFactory(id))
+	if err != nil {
+		if errHTTP, ok := err.(*requests.ErrorHTTP); ok && errHTTP.Code == 404 {
+			log.Info().Msgf("provider %q not have payout address for auto registration, will require manual registration", id.Address)
```

Comment:
Do we have to return here and not continue the registration? If I'm understanding this correctly new nodes will no longer auto register if we do this.

Reply:
This is how i recognise newly created identity, which have to be on-boarded manually via UI

Reply:
Ye it seems like you can do that and it works. Nice